### PR TITLE
Upgrade to the StructuredLogger 1.1.130.

### PR DESCRIPTION
**Pull request by KirillOsenkov**
Invoke the new DoubleWritesAnalyzer API and remove code that was doing the same thing.

This is a build infrastructure-only change.

### CODE REVIEW

File: build/scripts/test-build-correctness.ps1 [hunk 1]
```diff
@@ -25,15 +25,15 @@ try {
     # Need to parse out the current NuGet package version of Structured Logger
     $structuredLoggerPath = Join-Path (Get-PackageDir "Microsoft.Build.Logging.StructuredLogger") "lib\net46\StructuredLogger.dll"
     $configDir = Join-Path $binariesDir $config
-    $logPath = Join-Path $configDir "build.xml"
+    $logPath = Join-Path $configDir "roslyn.buildlog"
```
Comment (jaredpar) [inline build/scripts/test-build-correctness.ps1]: Can you update our groovy file to make sure we include buildlog files in the archiving process? Otherwise we won't be able to dig into this during failed jobs.

https://github.com/dotnet/roslyn/blob/master/netci.groovy#L21

File: build/scripts/test-build-correctness.ps1 [hunk 2]
```diff
@@ -25,15 +25,15 @@ try {
...
     $solution = Join-Path $repoDir "Roslyn.sln"
 
     if ($msbuild -eq "") {
         $msbuild = Ensure-MSBuild
     }
 
     Write-Host "Building Roslyn.sln with logging support"
-    Exec-Command $msbuild "/v:m /m /p:Configuration=$config /logger:StructuredLogger,$structuredLoggerPath;$logPath /nodeReuse:false /p:DeployExtension=false $solution"
+    Exec-Command $msbuild "/noconlog /v:m /m /p:Configuration=$config /logger:StructuredLogger,$structuredLoggerPath;$logPath /nodeReuse:false /p:DeployExtension=false $solution"
```
Comment (jaredpar) [inline build/scripts/test-build-correctness.ps1]: Why this change?

File: src/Tools/BuildBoss/StructuredLoggerCheckerUtil.cs [hunk 3]
```diff
@@ -1,106 +1,45 @@
...
         {
-            _document = document;
+            _logFilePath = logFilePath;
         }
 
         public bool Check(TextWriter textWriter)
         {
-            foreach (var element in _document.XPathSelectElements("//CopyTask"))
+            var build = Serialization.Read(_logFilePath);
+            var doubleWrites = DoubleWritesAnalyzer.GetDoubleWrites(build).ToArray();
```
Comment (jaredpar) [inline src/Tools/BuildBoss/StructuredLoggerCheckerUtil.cs]: You're making it too easy for me ;)

**Comment by jaredpar** [comment]
retest windows_debug_unit32_prtest please

**Comment by jaredpar** [comment]
retest windows_release_unit64_prtest please

**Comment by jaredpar** [comment]
retest windows_release_unit32_prtest please

File: build/scripts/test-build-correctness.ps1 (hunk 2 above)
Comment (KirillOsenkov) [inline build/scripts/test-build-correctness.ps1]: Removing the console output should speed the build up a bit, and we have the full log anyway if we need it. I can bring it back if you like?

File: build/scripts/test-build-correctness.ps1 (hunk 1 above)
Comment (KirillOsenkov) [inline build/scripts/test-build-correctness.ps1]: Fixed.

File: build/scripts/test-build-correctness.ps1 (hunk 2 above)
Comment (jaredpar) [inline build/scripts/test-build-correctness.ps1]: That's fine. Mostly wanted to understand why.

File: netci.groovy [hunk 4]
```diff
@@ -18,7 +18,8 @@ static void addRoslynJob(def myJob, String jobName, String branchName, Boolean i
   archiveSettings.addFiles('Binaries/**/*.dmp')
   archiveSettings.addFiles('Binaries/**/*.zip')
   archiveSettings.addFiles('Binaries/**/*.png')
-  archiveSettings.addFiles('Binaries/**/*.xml')
```
Comment (jaredpar) [inline netci.groovy]: Need to keep XML as it's the format for our unit test log

Comment (KirillOsenkov) [inline netci.groovy]: XML is already at the top of the list :)

Comment (jaredpar) [inline netci.groovy]: Indeed it is.

**Comment by KirillOsenkov** [comment]
Hmm, what is this failure:
https://ci.dot.net/job/dotnet_roslyn/job/master/job/windows_debug_vs-integration_prtest/5833/artifact/Binaries/Debug/UnitTests/VisualStudioIntegrationTests/xUnitResults/Screenshots/WorkspacesDesktop.ProjectReference-TargetInvocationException-09.46.02.log

**Comment by jaredpar** [comment]
retest windows_debug_vs-integration_prtest please

**Comment by jaredpar** [comment]
retest windows_release_vs-integration_prtest please

**Comment by jaredpar** [comment]
@KirillOsenkov doesn't look related. Sent a note to SWAT about that being potentially flaky. Kicking off another run to check.
